kernel/kernel-api/src/main/java/io/delta/kernel/types/TableSchemaSerDe.java

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.types;
+
+import static io.delta.kernel.types.PrimitiveType.BOOLEAN;
+import static io.delta.kernel.types.PrimitiveType.STRING;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import io.delta.kernel.client.JsonHandler;
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.data.ColumnarBatch;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.utils.Utils;
+
+/**
+ * Utility class to serialize and deserialize the table schema which is of type {@link StructType}.
+ */
+public class TableSchemaSerDe
```

Comment:
does this need be a public API?

Reply:
As of now, it needs to accessible to the `Scan.readData`. It needs to be public, but may be inside the `io.delta.kernel.internal`?